src/bin/cql-stress-cassandra-stress/java_generate/values/hex_blob.rs

```diff
@@ -0,0 +1,191 @@
+use super::ValueGenerator;
+use crate::java_generate::distribution::Distribution;
+
+/// Based on c-s HexBytes generator.
+/// Used by c-s to generate blob partition keys.
+/// TBH, not quite sure why c-s won't use Bytes generator for this.
+pub struct HexBlob;
+
+impl ValueGenerator for HexBlob {
+    type Value = Vec<u8>;
+
+    fn generate(
+        &mut self,
+        identity_distribution: &mut dyn Distribution,
+        size_distribution: &mut dyn Distribution,
+    ) -> Self::Value {
+        let seed = identity_distribution.next_i64();
+        size_distribution.set_seed(seed);
+        let size = size_distribution.next_i64() as usize;
+
+        let mut result = vec![0; size];
+        let mut i = 0;
+
+        // No idea what's going on here.
+        // Check https://github.com/scylladb/scylla-tools-java/blob/master/tools/stress/src/org/apache/cassandra/stress/generate/values/HexBytes.java#L44
+        while i < size {
+            let mut value = identity_distribution.next_i64();
+            let mut j = 0;
+            while j < 16 && i + j < size {
+                let v = (value & 15) as i32;
+                result[i + j] = ((if v < 10 { '0' as i32 } else { 'A' as i32 }) + v) as u8;
+                // Unsigned right shift.
+                value = ((value as u64) >> 4) as i64;
+                j += 1;
+            }
```

Comment:
A hexadecimal digit is equal to 4 bits. `value` has 64 bits, and 64 = 4 * 16.

This code just takes a i64 random number (`value`) and generates at most 16 hexadecimal digits from it: takes 4 lowest bits (`let v = (value & 0xF) as i32`), writes a hexadecimal digit and finally discards the lowest 4 bits, shifting to the right (`value = ((value as u64) >> 4) as i64`).